cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -1117,6 +1129,35 @@ protected void jvmPerformanceOptions(List<EnvVar> envVars) {
         return certs;
     }
 
+    /**
+     * Generate the OwnerReference object to link newly created objects to their parent (the custom resource)
+     *
+     * @return
+     */
+    protected OwnerReference createOwnerReference() {
+        return new OwnerReferenceBuilder()
+                .withApiVersion(ownerApiVersion)
+                .withKind(ownerKind)
+                .withName(cluster)
+                .withUid(ownerUid)
+                .withBlockOwnerDeletion(false)
+                .withController(false)
+                .build();
+    }
+
+    /**
+     * Set fields needed to generate the OwnerReference object
+     *
+     * @param apiVersion    API version of the parent object
+     * @param kind          Kind of the parent object
+     * @param uid           UID of the parent object
+     */
+    protected void setOwnerReference(String apiVersion, String kind, String uid)  {
```

Comment:
Wouldn't it be more natural just to accept a `HasMetadata` instance here, rather than forcing callers to decompose the CR they already have an instance of?

Reply:
TBH, so far we always use the original CR only the `fromCrd` method and do not pass it to any other method. That is why I implemented it as I did. But I can for sure change that if you think its ok.

Reply:
This should be fixed now.